railties/lib/rails/generators/rails/app/templates/Gemfile

```diff
@@ -21,6 +21,11 @@ source 'https://rubygems.org'
 # Use Capistrano for deployment
 # gem 'capistrano-rails', group: :development
 
+<%- if options.api? -%>
+# Use ActiveModelSerializers to serialize JSON responses
+gem 'active_model_serializers', '~> 0.10.0.rc1'
```

Comment:
The implementation of each serializer is basically describing what you need to be serialized in a declarative way. You're not defining a format. You just say serialize :title, :body, with associations with :comments, :author, etc and you get a thing formatted for instance in JSON API way http://jsonapi.org/ and I can even change to another format without changing the code.

If you were doing that with a template based library you need to explicitly put all the code to generate each tag, attribute, value, etc and once you put that code you can't change the format.
And it's also error prone because you can generate an invalid format.